Formula/func-e.rb

```diff
@@ -0,0 +1,21 @@
+class FuncE < Formula
+  desc "Easily run Envoy®"
+  homepage "https://func-e.io"
+  url "https://github.com/tetratelabs/func-e/archive/v0.5.0.tar.gz"
+  sha256 "17fc6c5c0f7bea8ce59e0bfb315198cc9f9ecfda98d51909149d42ffd125d72f"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/tetratelabs/func-e/internal/version.funcE=#{version}
+    ]
+    system "go", "build", "-trimpath", "-ldflags", ldflags.join(" "), "-o", bin/"func-e"
+  end
+
+  test do
+    system "#{bin}/func-e", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
sure no problem. I will make a test for this output when done dinner. The versions will change, but there will always be at least one (pointers welcome, but I think I can figure it out.)

```bash
$ func-e versions -a
  1.18.3 2021-05-11
  1.17.3 2021-05-11
  1.16.4 2021-05-11
  1.15.5 2021-05-11
  1.14.7 2021-04-15
  1.13.7 2020-12-07
```

Reply:
okie dokie. PTAL as I think the test is much better!

Reply:
also I'm not sure it is ok to do it, but we could make this pass if rosetta is installed after this change. https://github.com/tetratelabs/func-e/pull/300

Note this would be a work-around as the underlying constraint in envoy is still there..